python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -8191,3 +8191,85 @@ def test_agg_for_dataframe_with_string_columns(aggs):
         ),
     ):
         gdf.agg(aggs)
+
+
+@pytest.mark.parametrize(
+    "join", ["left"],
+)
+@pytest.mark.parametrize(
+    "overwrite", [True, False],
+)
+@pytest.mark.parametrize(
+    "filter_func", [None],
+)
+@pytest.mark.parametrize(
+    "errors", ["ignore"],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [1, 2, 3], "b": [3, 4, 5]},
+        {"e": [1.0, 2.0, 3.0], "d": [3.0, 4.0, 5.0]},
+        {"b": [4.0, 3.0, 1.0], "c": [2.0, 5.0, 8.0]},
+        {"c": [True, False, False], "d": [False, True, True]},
+        {"g": [2.0, np.nan, 4.0], "n": [np.nan, np.nan, np.nan]},
+        {"b": [np.nan, np.nan, 3.0], "c": [6.0, np.nan, np.nan]},
+        {"d": [np.nan, np.nan, np.nan], "e": [np.nan, np.nan, np.nan]},
+        {"b": [np.nan, np.nan, np.nan], "c": [np.nan, np.nan, np.nan]},
+    ],
+)
+@pytest.mark.parametrize(
+    "data2",
+    [
+        {"b": [3, 5, 6], "e": [8, 2, 1]},
+        {"a": [7, 5, 8], "b": [2.0, 7.0, 9.0]},
+        {"c": [True, False, True], "d": [3.0, 4.0, 5.0]},
+        {"e": [False, False, True], "g": [True, True, False]},
+        {"a": [np.nan, np.nan, np.nan], "b": [np.nan, np.nan, np.nan]},
+        {"b": [np.nan, 2.0, np.nan], "c": [2, np.nan, 5.0]},
+    ],
+)
+def test_update_for_dataframes(
```

Comment:
@skirui-source I think we have added different columns testcases but we still need to add test cases where we cover the no-equal index cases too.